server/src/main/java/io/druid/client/cache/CacheConfig.java

```diff
@@ -70,6 +70,6 @@ public int getCacheBulkMergeLimit()
   public boolean isQueryCacheable(Query query)
   {
     // O(n) impl, but I don't think we'll ever have a million query types here
-    return !unCacheable.contains(query.getType());
+    return !query.isDescending() && !unCacheable.contains(query.getType());
```

Comment:
Yes, I should have done some homework on cache but apparently I've forgot that. I'll check that.